packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -294,11 +295,38 @@ const ProductDetail = () => {
                 }
             ]
 
-            await addItemToBasketMutation.mutateAsync({
+            const res = await addItemToBasketMutation.mutateAsync({
                 parameters: {basketId: basket.basketId},
                 body: productItems
             })
 
+            // the response from addItemToBasketMutation returns an entire basket including newly added bundle product along with other products in the cart
+            // there is no reliable way to filter out the bundle among productItems
+            // Hence, we have to loop over each basket items, and its bundleProductItems to search out the children that still have
+            // masterId and update it with the variantId
+            res.productItems.forEach((productItem) => {
```

Comment:
I was messing around with the API, it seems that the most recent item added to the basket will be added to the end of the `productItems` array. I haven't confirmed with documentation/baskets team, but we should check first cause ideally we won't want to run this bundles logic for all product bundles in the basket cause it may be a source for bugs down the line